src/lib/storageManager.ts

```
@@ -0,0 +1,432 @@
+   * Parse and validate storage entry
+   */
+  private parseStorageEntry(data: string): { value: string; isValid: boolean } {
+    try {
+      const entry = JSON.parse(data);
+      
+      // Validate structure
+      if (!entry || typeof entry !== 'object' || !('value' in entry)) {
+        return { value: data, isValid: false }; // Fallback to raw data
+      }
+
+      let value = entry.compressed ? this.decompress(entry.value) : entry.value;
```

ESLint

[error] 'value' is never reassigned. Use 'const' instead.

(prefer-const)